python/lsst/pipe/tasks/healSparseMapping.py

```diff
@@ -0,0 +1,274 @@
+        try:
+            data = hsp.HealSparseMap.read(path, pixels=pixels, degrade_nside=degrade_nside)
+        except (OSError, RuntimeError):
+            raise ValueError(f"Unable to read healsparse map with URI {self.fileDescriptor.location.uri}")
+
+        return data
+
+    def write(self, inMemoryDataset):
+        # Docstring inherited from Formatter.write.
+        # Update the location with the formatter-preferred file extension
+        self.fileDescriptor.location.updateExtension(self.extension)
+        inMemoryDataset.write(self.fileDescriptor.location.path)
+
+
+class HealSparseInputMapConfig(pexConfig.Config):
+    """Configuration parameters for HealSparseInputMapTask"""
+    nside = pexConfig.Field(
+        doc="Mapping healpix nside.  Must be power of 2.",
+        dtype=int,
+        default=32768,
+    )
+    nside_coverage = pexConfig.Field(
+        doc="HealSparse coverage map nside.  Must be power of 2.",
+        dtype=int,
+        default=256,
+    )
+    bad_mask_min_coverage = pexConfig.Field(
+        doc=("Minimum area fraction of a map healpixel pixel that must be "
+             "covered by bad pixels to be removed from the input map. "
+             "This is approximate."),
+        dtype=float,
+        default=0.5,
+    )
+
+
+class HealSparseInputMapTask(pipeBase.Task):
+    """Task for making a HealSparse input map."""
+
+    ConfigClass = HealSparseInputMapConfig
+    _DefaultName = "healSparseInputMap"
+
+    def __init__(self, **kwargs):
+        pipeBase.Task.__init__(self, **kwargs)
+
+        self.ccd_input_map = None
+
+    def build_ccd_input_map(self, bbox, wcs, ccds):
+        """Build a map from ccd valid polygons or bounding boxes.
+
+        Parameters
+        ----------
+        bbox : `lsst.geom.Box2I`
+            Bounding box for region to build input map.
+        wcs : `lsst.afw.geom.SkyWcs`
+            WCS object for region to build input map.
+        ccds : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with ccd data from coadd inputs.
+        """
+        self.ccd_input_map = hsp.HealSparseMap.make_empty(nside_coverage=self.config.nside_coverage,
+                                                          nside_sparse=self.config.nside,
+                                                          dtype=hsp.WIDE_MASK,
+                                                          wide_mask_maxbits=len(ccds))
+        self._wcs = wcs
+        self._bbox = bbox
+        self._ccds = ccds
+
+        pixel_scale = wcs.getPixelScale().asArcseconds()
+        hpix_area_arcsec2 = hp.nside2pixarea(self.config.nside, degrees=True)*(3600.**2.)
+        self._min_bad = self.config.bad_mask_min_coverage*hpix_area_arcsec2/(pixel_scale**2.)
+
+        metadata = {}
+        self._bits_per_visit_ccd = {}
+        self._bits_per_visit = defaultdict(list)
+        for bit, ccd in enumerate(ccds):
+            metadata[f'B{bit:04d}CCD'] = ccd['ccd']
```

Comment:
This is defined here: https://github.com/lsst/pipe_tasks/blob/a397b918da8467468a11e6a32442581973bd1148/python/lsst/pipe/tasks/coaddInputRecorder.py#L173 and I know that I've mentioned this on slack.  Right now it's hard-coded to be called `"ccd"` and if it isn't a whole bunch of code is going to break (including this code!).  I don't know of any plans to update this to `"detectorId"` (it can't be `"detector"` because that's reserved for the `ExposureCatalog`).